src/Client.php

```diff
@@ -290,6 +290,153 @@ public function upload(string $path, $contents, $mode = 'add'): array
     }
 
     /**
+     * Upload file split in chunks. This allows uploading large files, since
+     * Dropbox API v2 limits the content size to 150MB.
+     *
+     * The chunk size will affect directly the memory usage, so be careful.
+     * Large chunks tends to speed up the upload, while smaller optimizes memory usage.
+     *
+     * @param string          $path
+     * @param string|resource $contents
+     * @param int             $chunkSize
+     * @param string          $mode
+     *
+     * @return array
+     */
+    public function uploadChunked(string $path, $contents, int $chunkSize, $mode = 'add'): array
+    {
+        // This method relies on resources, so we need to convert strings to resource
+        if (is_string($contents)) {
+            $stream = fopen('php://memory', 'r+');
+            fwrite($stream, $contents);
+            rewind($stream);
+        } else {
+            $stream = $contents;
+        }
+
+        $data = self::readFully($stream, $chunkSize);
+        $cursor = null;
+
+        while (! ((strlen($data) < $chunkSize) || feof($stream))) {
+            // Start upload session on first iteration, then just append on subsequent iterations
+            if (isset($cursor)) {
+                $cursor = $this->uploadSessionAppend($data, $cursor);
+            } else {
```

Comment:
I really dislike `else` statements. Could you replace this by using a ternary operator?